### DISCUSSION TITLE
The initial execution of 'sam sync' demands user interaction

### DISCUSSION THREAD

**Discussion by gusth-sa** [aws/aws-sam-cli#6694] [Ideas]
When configuring a new stack, to start a new sync process is necessary to confirm the following prompt

```
The SAM CLI will use the AWS Lambda, Amazon API Gateway, and AWS StepFunctions APIs to upload your code without 
performing a CloudFormation deployment. This will cause drift in your CloudFormation stack. 
**The sync command should only be used against a development stack**.

Confirm that you are synchronizing a development stack.

Enter Y to proceed with the command, or enter N to cancel:
 [Y/n]: 
```

After proceeding with 'Y', an update is made to `~/.aws-sam/metadata.json`, appending a new value to the `accelerateOptInStacks` array, and subsequent sync operations will not prompt this action again. 

I tried to identify this value but seems to be an MD5 that I could not relate to any stack information.

Though understandable for safety, there's a need for an option in 'sam sync' to designate development stacks and skip prompts, especially in automated setups.

**Comment by ryanhitt40** [comment]
Yeah agree it should have an option to skip this. However if you need to automate it in jenkins/whatever just echo the Y and pipe it to the sync command
